environment/deployments/science-platform/alertdb/main.tf

```diff
@@ -0,0 +1,111 @@
+# Creates two buckets, one for alert packet data and one for alert schema data.
+
+# The alert packet bucket may optionally have a lifecycle rule to delete old
+# alert data.
+
+# Creates two ServiceAccounts, one with readwrite access and the other with
+# readonly access to the objects. Sets both up to be accessible from
+# Kubernetes via Workload Identity.
+
+module "alert_packet_bucket" {
+  source        = "../../../../modules/bucket"
+  project_id    = var.project_id
+  storage_class = "REGIONAL"
+  location      = "us-central1"
+  prefix_name   = "alertdb"
+  suffix_name   = ["packets"]
+  labels        = var.labels
+
+  # Conditionally set up a rule to delete old alert data
+  lifecycle_rules = var.purge_old_alerts ? [
+    {
+      action = {
+        type = "Delete"
+      },
+      condition = {
+        age = var.maximum_alert_age
```

Comment:
Age has one-day granularity.  Should this be >= ?  Does this necessarily get run each day?

Reply:
Yes, this is correct as-is. 

The way this works is through [Object Lifecycle Management](https://cloud.google.com/storage/docs/lifecycle#age). It's a setting you can apply to a bucket which instructs Google to take care of object cleanup on your behalf. The "age" field acts like you would hope:

> For example, if an object's creation time is 2019/01/10 10:00 UTC and the Age condition is 10 days, then the condition is satisfied for the object on **and after** 2019/01/20 10:00 UTC. 

(emphasis added)

A second reason that `>=` isn't permitted is that, despite the appearance, this `condition` value is a Terraform `map` type - just a `string` -> `anything` mapping, equivalent to the Python `{"age": maximum_alert_age}`. Using `>=` would be a syntax error here.